crates/superposition_types/src/custom_query.rs

```diff
@@ -86,3 +91,108 @@ where
         ready(Self::extract_query(req.query_string()).map_err(actix_web::Error::from))
     }
 }
+
+/// Provides struct to extract those query params from the request which are `not wrapped` in contrusts like `platform[param_name]`
+#[derive(Debug, Clone)]
+pub struct DynamicQuery<T: DeserializeOwned>(pub T);
+
+impl<T> CustomQuery for DynamicQuery<T>
+where
+    T: DeserializeOwned,
+{
+    type Inner = T;
+
+    fn regex_pattern() -> &'static str {
+        r"^(?<query_name>[^\[\]]+)$"
+    }
+
+    fn capture_group() -> &'static str {
+        "query_name"
+    }
+
+    fn into_inner(self) -> T {
+        self.0
+    }
+
+    fn new(inner: Self::Inner) -> Self {
+        Self(inner)
+    }
+}
+
+impl<T> actix_web::FromRequest for DynamicQuery<T>
+where
+    T: DeserializeOwned,
+{
+    type Error = actix_web::Error;
+    type Future = std::future::Ready<Result<Self, Self::Error>>;
+
+    fn from_request(
+        req: &actix_web::HttpRequest,
+        _: &mut actix_web::dev::Payload,
+    ) -> Self::Future {
+        use std::future::ready;
+        ready(Self::extract_query(req.query_string()).map_err(actix_web::Error::from))
+    }
+}
+
+/// Provides struct to `Deserialize` `HashMap<String, String>` as `serde_json::Map<String, serde_json::Value>`
+#[derive(Deserialize)]
+#[serde(from = "HashMap<String,String>")]
+pub struct QueryMap(Map<String, Value>);
```

Comment:
Is there a reason we want serde::Value here? Why not string? You can use that in Query directly:
https://users.rust-lang.org/t/actix-web-accessing-query-parameters-untyped-ly/50341